src/setfit/modeling.py

```diff
@@ -677,19 +677,21 @@ def sentence_pairs_generation(sentences, labels, pairs):
     # labels to indicate if a pair is positive or negative
 
     num_classes = np.unique(labels)
-    idx = [np.where(labels == i)[0] for i in num_classes]
+    label_to_idx = {x: i for i, x in enumerate(num_classes)}
+    positive_idxs = [np.where(labels == i)[0] for i in num_classes]
+    negative_idxs = [np.where(labels != i)[0] for i in num_classes]
 
-    for first_idx in range(len(sentences)):
+    for first_idx in tqdm(range(len(sentences)), desc="Gen Pairs"):
```

Comment:
I have a few comments here:
1. Can we use [`trange(...)`](https://tqdm.github.io/docs/tqdm/#trange) instead of `tqdm(range(...))`?
2. Perhaps `Generating Training Pairs` or `Generating Contrastive Training Pairs` is a bit clearer than `Gen Pairs`?
3. Can we use [`leave=False`](https://github.com/tqdm/tqdm#parameters)? Otherwise we end up with e.g. 20 completed `tqdm` bars after the pairs are generated.

Reply:
I totally agree! And actually, it's a little weird/confusing to show the progress for each iteration. 

Instead, I moved the `trange` into `trainer.py`, so the user just sees the top-level progress for all iterations being generated